### ISSUE TITLE
question: Why does `doveadm auth test` fails with user added by `setup dovecot-master add` and `setup email add`

### ISSUE THREAD

**Issue by unlucio** [docker-mailserver/docker-mailserver#4098]
### Subject

Something else that requires developers attention

### Description

while still trying to debug this problem https://github.com/orgs/docker-mailserver/discussions/4096

looks like dovecot doesn't like users added by setup:

I'm using the `setup` command as suggested by the documentation but:

```
setup dovecot-master add test@mydomain.com aaaa
doveadm auth test test@mydomain.com aaaa
passdb: test@mydomain.com auth failed
extra fields:
  user=test@mydomain.com

setup email add test@mydomain.com aaaa 
doveadm auth test test@mydomain.com aaaa
passdb: test@mydomain.com auth failed
extra fields:
  user=test@mydomain.com
```

is this supposed to happen? 😵‍💫

**Comment by polarathene** [comment]
If tat wasn't working our test suite would fail:

https://github.com/docker-mailserver/docker-mailserver/blob/9175424d0f3c67622c030311cc720d7477f1c2b6/test/tests/parallel/set3/scripts/setup_cli.bats#L40-L67

It's likely that you re running the `doveadm` command too soon?

What happens is the `setup email add` command updates the `postfix-accounts.cf` file with a new line, and within 2 seconds that change should be detected which triggers Dovecot to add the account to it's UserDB. After all the change detection processing is done it issues a `dovecot reload`, from here your `doveadm` should work.

As you can see in the test case we wait until the account is created before we check auth. In the past this could take around 10s, but it should be fairly quick now, the 2 second window is how often the change detection service polls config files for changes.

**Comment by unlucio** [comment]
I doubt it was too soon:
my 1st test was actually with an account I created yesterday during setup, I'm still investigating why everything seems fine but my logs are full of those not-so-helpful errors and my user cannot authenticate. That container has been destroyed and created a ton of times right now.
I created the test accounts just as props for this post, after I saw `doveadm auth test` systematically fail.
To be sure I just ran that auth test again, several hours later, and here's the result:
<img width="377" alt="Screenshot_2024-07-04_at_00_23_57" src="https://github.com/docker-mailserver/docker-mailserver/assets/778799/5980c6ed-8d98-4b93-b5c4-cbb64141e30a">

```
root@mail:/# setup email list
* postmaster@mydomain.com

* test@mydomain.com

root@mail:/# setup dovecot-master list
* postmaster@mydomain.com

* test@mydomain.com

root@mail:/# doveadm auth test test@mydomain.com aaaaa
passdb: test@mydomain.com auth failed
extra fields:
  user=test@mydomain.com
```

I see the expected account,m and it still fails, both the test as well if I try a mail client.

**Comment by polarathene** [comment]
> while still trying to debug this problem https://github.com/orgs/docker-mailserver/discussions/4096

I've since read and responded to your related discussion.

Without a full bug report I've had to make some assumptions, but I'm going to guess you're trying to run DMS with Docker on macOS. That uses a VM with macOS specific drivers that are known to have issues, you can try searching through issues we've tagged with macOS label, or find the upstream issues on Github unrelated to DMS.

I know I've had no issue with this on Linux hosts or in WSL on Windows. I was able to authenticate without trouble with `doveadm auth test` or via ThunderBird.

I haven't done so recently, but given our test suite passes here I really don't see why that would fail for me if I were to try again.

---

> I see the expected account, and it still fails, both the test as well if I try a mail client.

If `doveadm auth test` fails, I don't think the mail client would succeed. You need to get that command working first.

If I am right about you trying to make DMS work with Docker on macOS, please try reproduce in an actual Linux VM running Docker without those macOS specific concerns in the way.

Here on WSL2 (Windows 11) your issue cannot be reproduced:

```yml
services:
  test:
    image: mailserver/docker-mailserver:edge
    container_name: dms
    hostname: example.test
```

```console
$ docker compose up -d
$ docker exec -it dms bash

# In the DMS container now:
$ setup email add hello@example.test secret
$ doveadm auth test hello@example.test secret
passdb: hello@example.test auth succeeded
extra fields:
  user=hello@example.test
```

**Comment by unlucio** [comment]
Thank you for your prompt answer, unfortunately, I ended up on an unplanned "vacation" in the hospital shortly after.

I'm not running DMS on macOS but in a docker-swarm cluster (all linux machines), forcing the deploy on the master node for sake of not having to deal too much with eventual network-related problems (the swarm ingress is actually traefik and ideally DMS will be behind traefik at some point).
The master-node, and machine where DMS is currently running, is:
- distro: Ubuntu 22.04.4 LTS
- kernel: 5.15.0-113-generic  123-Ubuntu SMP Mon Jun 10 08:16:17 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux
- docker: Docker version 20.10.17, build 100c701

> If doveadm auth test fails, I don't think the mail client would succeed. You need to get that command working first.

I completely agree, which is the reason I ditched the client (mail.app was just the most convenient one at the time of testing), and focused on that.

All what I reported was indeed on Linux.

I agree that having a full trace to debug it would be best: I tried to provide thought information about my setup on my community post. I hoped my environment, logs, and config files would be a good start. I'll add my compose as well:

```yaml
version: '3.8'

services:
  mailserver:
    image: ghcr.io/docker-mailserver/docker-mailserver:latest
    hostname: mail.mydomain.com
    # Provide the FQDN of your mail server here (Your DNS MX record should point to this value)
    env_file: .env
    # More information about the mail-server ports:
    # https://docker-mailserver.github.io/docker-mailserver/latest/config/security/understanding-the-ports/
    # To avoid conflicts with yaml base-60 float, DO NOT remove the quotation marks.
    ports:
      - '25:25' # SMTP  (explicit TLS => STARTTLS, Authentication is DISABLED => use port 465/587 instead)
      - '143:143' # IMAP4 (explicit TLS => STARTTLS)
      - '465:465' # ESMTP (implicit TLS)
      - '587:587' # ESMTP (explicit TLS => STARTTLS)
      - '993:993' # IMAP4 (implicit TLS)
    volumes:
      - /etc/localtime:/etc/localtime:ro
      - /mnt/data/docker-mailserver/mail-data/:/var/mail/
      - /mnt/data/docker-mailserver/mail-state/:/var/mail-state/
      - /mnt/data/docker-mailserver/mail-logs/:/var/log/mail/
      - /mnt/data/docker-mailserver/config/:/tmp/docker-mailserver/
      - /mnt/data/traefik/certs/:/etc/letsencrypt/
    networks:
      - swarm-net
    deploy:
      mode: replicated
      replicas: 1
      placement:
        constraints:
          - node.role == manager
    # Uncomment if using `ENABLE_FAIL2BAN=1`:
    # cap_add:
    #   - NET_ADMIN
    healthcheck:
      test: "ss --listening --tcp | grep -P 'LISTEN.+:smtp' || exit 1"
      timeout: 3s
      retries: 0

networks:
  swarm-net:
    external: true
```

What else can I dig up to provide useful information?
I of course thought about debug logs but they might be too much?

**Comment by polarathene** [comment]
> I would have done a bug report if I found a bug, but as for now, I don't feel like I found one. I start assuming I did something wrong before assuming it is a bug.

That's fine, I was specifically interested in the information we'd otherwise have if you did fill out a bug report. It provides what we want to know up front in a structured way and we can then proceed as we are now.

If maintainers then deem it is not a bug we could close it, but that's really no different to raising an issue that avoids filling out a proper bug report.

---

> I'm not running DMS on macOS but in a docker-swarm cluster (all linux machines), forcing the deploy on the master node for sake of not having to deal too much with eventual network-related problems (the swarm ingress is actually traefik and ideally DMS will be behind traefik at some point). The master-node, and machine where DMS is currently running, is:
> 
> * distro: Ubuntu 22.04.4 LTS
> * kernel: 5.15.0-113-generic  123-Ubuntu SMP Mon Jun 10 08:16:17 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux
> * docker: Docker version 20.10.17, build 100c701

Thank you.

Your kernel version should be ok, but your [Docker Engine is 2 years old](https://docs.docker.com/engine/release-notes/20.10/#201017) and the newer major releases since bring notable improvements and bug fixes. You're probably on Docker Compose 1.x instead of 2.0 too. Visit the official docs for Docker to get install advice for upgrading these two.

Traefik / routing aside, we do not support Docker Swarm.

Could you please confirm that your issue still occurs with [`compose.yaml` like with our example](https://github.com/docker-mailserver/docker-mailserver/blob/34423c2f665b8c7839c40fbc7d1fe8cda6342456/compose.yaml), no Docker Swarm involved.

We need a reproduction with that, and I suspect your issue is most likely Docker Swarm specific, or potentially with the outdated Docker software.

---

> `version: '3.8'`

You won't need this once you use Docker Compose v2.

> `image: ghcr.io/docker-mailserver/docker-mailserver:latest`

You should also at least pin to major version, otherwise you may unintentionally update to a new major version without having read the changelog in advance.

> `- /mnt/data/docker-mailserver/mail-state/:/var/mail-state/`

While you're troubleshooting this, you might want to try disabling this volume, just comment it out. Ensure that you use `docker compose up --force-recreate` when you make changes and want to restart DMS too.

---

> What else can I dig up to provide useful information?

Minimize your `compose.yaml`, definitely attempt to reproduce without Docker Swarm. You want to simplify to the point that you can still reproduce the problem.

Since this is just about `doveadm auth test` as your reproduction verifier, you really shouldn't need much for a minimal reproduction:

```console
$ docker run --rm -it --hostname mail.example.test --name dms mailserver/docker-mailserver:edge

# In another terminal, shell into the DMS container to attempt reproduction:
$ docker exec -it dms bash
$ setup email add hello@example.test secret
$ doveadm auth test hello@example.test secret
passdb: hello@example.test auth succeeded
extra fields:
  user=hello@example.test
```

As you can see on my system, there is no problem. I should also re-read my previous responses since I already demonstrated this with `compose.yaml` instead of `docker run`.

If you cannot reproduce the same on your end, you'll need to iteratively add back config until you find out what triggers it. If you cannot reproduce it without Docker Swarm, then it'll be an incompatibility with Docker Swarm and no maintainer will be able to assist troubleshooting that.

**Comment by unlucio** [comment]
> Your kernel version should be ok, but your [Docker Engine is 2 years old](https://docs.docker.com/engine/release-notes/20.10/#201017) and the newer major releases since bring notable improvements and bug fixes. You're probably on Docker Compose 1.x instead of 2.0 too. Visit the official docs for Docker to get install advice for upgrading these two.

Yeah, even tho I originally installed it following docker's instruction, it should use docker's source repos, idk (but I haven't really investigated) why it's stuck on such an old version. It didn't bother me too much but now it's turning into a pain in the butt for a number of reasons so I'll rebuild the cluster probably this weekend.

I took the chance to remake the cluster and migrate to kube (k3s) (does DMS support kube?).
Now I'm running :
```
masterk ~ # cat /etc/os-release
PRETTY_NAME="Ubuntu 24.04 LTS"
NAME="Ubuntu"
VERSION_ID="24.04"
VERSION="24.04 LTS (Noble Numbat)"
VERSION_CODENAME=noble
ID=ubuntu
ID_LIKE=debian
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
UBUNTU_CODENAME=noble
LOGO=ubuntu-logo
masterk ~ # docker --version
Docker version 27.1.1, build 6312585
masterk ~ # docker compose version
Docker Compose version v2.29.1
```
Having freshly installed machines I took the chance to run DMS before configuring k3s, but unfortunately the situation didn't change:

```
root@mail:/# doveadm auth test test@mydomain.com aaaa
passdb: test@mydomain.com auth failed
extra fields:
  user=test@mydomain.com
root@mail:/# doveadm auth test postmaster@mydomain.com aaaa
passdb: postmaster@mydomain.com auth failed
extra fields:
  user=postmaster@mydomain.com
```

```
mailserver-1  | 2024-07-29T14:54:22.614128+02:00 mail dovecot: auth: passwd-file(test@mydomain.com): unknown user (SHA1 of given password: d012f6)
mailserver-1  | 2024-07-29T14:54:36.777718+02:00 mail dovecot: auth: passwd-file(postmaster@mydomain.com): unknown user (SHA1 of given password: d012f6)
```

I also tried creating and testing a new user:
```
root@mail:/# setup email add test2@mydomain.com aaaa
root@mail:/# setup dovecot-master add test2@mydomain.com aaaa
root@mail:/# doveadm auth test test2@mydomain.com aaaa
passdb: test2@mydomain.com auth failed
extra fields:
  user=test2@mydomain.com
```
but the situation didn't change:
```
mailserver-1  | 2024-07-29T15:28:01.602462+02:00 mail postfix/postfix-script[6134]: refreshing the Postfix mail system
mailserver-1  | 2024-07-29T15:28:01.615887+02:00 mail postfix/master[1118]: reload -- version 3.7.10, configuration /etc/postfix
mailserver-1  | 2024-07-29T15:28:01.634118+02:00 mail dovecot: master: Warning: SIGHUP received - reloading configuration
mailserver-1  | 2024-07-29T15:28:26.123214+02:00 mail postfix/postfix-script[6330]: refreshing the Postfix mail system
mailserver-1  | 2024-07-29T15:28:26.135747+02:00 mail postfix/master[1118]: reload -- version 3.7.10, configuration /etc/postfix
mailserver-1  | 2024-07-29T15:28:26.153751+02:00 mail dovecot: master: Warning: SIGHUP received - reloading configuration

mailserver-1  | 2024-07-29T15:28:40.633577+02:00 mail dovecot: auth: passwd-file(test2@mydomain.com): unknown user (SHA1 of given password: d012f6)
```

I also noticed these errors:
```
mailserver-1  | 2024-07-29T15:28:40.633135+02:00 mail dovecot: auth: Error: passwd-file /etc/dovecot/userdb:User postmaster@mydomain.com has invalid UID '0'
mailserver-1  | 2024-07-29T15:28:40.633154+02:00 mail dovecot: auth: Error: passwd-file /etc/dovecot/userdb:User test@mydomain.com has invalid UID '0'
mailserver-1  | 2024-07-29T15:28:40.633164+02:00 mail dovecot: auth: Error: passwd-file /etc/dovecot/userdb:User test2@mydomain.com has invalid UID '0'
```
But the files are there and they have the expected data:
```
root@mail:/# cat /etc/dovecot/userdb | wc -l
3
```

> Could you please confirm that your issue still occurs with [`compose.yaml` like with our example](https://github.com/docker-mailserver/docker-mailserver/blob/34423c2f665b8c7839c40fbc7d1fe8cda6342456/compose.yaml), no Docker Swarm involved.

Yes, it does.

> We need a reproduction with that, and I suspect your issue is most likely Docker Swarm specific, or potentially with the outdated Docker software.

docker swarm is gone, now I'm testing it on a single, updated, machine but it doesn't seem to help.

> `version: '3.8'`
removed that one too, no change

> 
> You won't need this once you use Docker Compose v2.
> 
> > `image: ghcr.io/docker-mailserver/docker-mailserver:latest`
> 
> You should also at least pin to major version, otherwise you may unintentionally update to a new major version without having read the changelog in advance.

I do that once things work, while I'm still deciding if to keep them I like to have the latest version. That's just a matter of preferences, I'm not continuously deleting and fetching the image, so no chances of unexpected changes.
Not my 1st rodeo using computers ;)


> > `- /mnt/data/docker-mailserver/mail-state/:/var/mail-state/`
> While you're troubleshooting this, you might want to try disabling this volume, just comment it out. Ensure that you use `docker compose up --force-recreate` when you make changes and want to restart DMS too.
I was deleting the container every time anyway. Commenting that mounting point didnìt help either.

> Minimize your `compose.yaml`, definitely attempt to reproduce without Docker Swarm. You want to simplify to the point that you can still reproduce the problem.

done, but same behaviour

> Since this is just about `doveadm auth test` as your reproduction verifier, you really shouldn't need much for a minimal reproduction:
> 
> ```
> $ docker run --rm -it --hostname mail.example.test --name dms mailserver/docker-mailserver:edge
> 
> # In another terminal, shell into the DMS container to attempt reproduction:
> $ docker exec -it dms bash
> $ setup email add hello@example.test secret
> $ doveadm auth test hello@example.test secret
> passdb: hello@example.test auth succeeded
> extra fields:
>   user=hello@example.test
> ```

This actually succeded:
```
masterk ~ # docker exec -it 71f83984e286 bash
root@mail:/# setup email add hello@example.test secret
root@mail:/# doveadm auth test hello@example.test secret
Error: auth-client: conn unix:/run/dovecot/auth-client: connect(/run/dovecot/auth-client) failed: No such file or directory
root@mail:/# doveadm auth test hello@example.test secret
passdb: hello@example.test auth succeeded
extra fields:
  user=hello@example.test
root@mail:/#
```
One of the main differences I noticed is the one-liner uses the `edge` tag, so I tried using that in my compose file, but no joy.
The 2nd difference is that in my tastes I was always using the same data, so I created new empty directories and tried again: still no joy
The 3rd possible difference is that the data directory is an `nfs` share, so I tried with a local directory: still no joy.

And now I'm out of options.



> As you can see on my system, there is no problem. I should also re-read my previous responses since I already demonstrated this with `compose.yaml` instead of `docker run`.
`docker run` works for me as well, while `docker compose` systematically fails no matter what I do. Even the "distro" one.

>> If you cannot reproduce the same on your end, you'll need to iteratively add back config until you find out what triggers it. >If you cannot reproduce it without Docker Swarm, then it'll be an incompatibility with Docker Swarm and no maintainer will be able to assist troubleshooting that.

In light of the same issue presenting its self even on a bran new install ona  single host, In all honesty I think it's hardly wver bee`docker swarm'`s fault.
So the most likely culprit here is my env configuration, isn't it? (reason why I posted it all at the beginning)
This project has a ton of options (not a bad thing, just pretty incovienient to test the all when one is new to a project), what would be a minum viable env I can use to see if at least with that works.

**Comment by polarathene** [comment]
> I took the chance to rebuild the cluster and migrate to kube (k3s) (does DMS support kube?).

A maintainer uses k8s, and we do have a docs page for k8s related config, along with a separate helm chart repo. There is no official support though, and that maintainer is very limited on time. Consider it equivalent to community contribution, it's known to work but you'll likely only find support from other community members.

I don't have k8s experience myself, so I don't know how that differs with kube (_I am vaguely familiar with what that is, but that's it_).

---

> I also noticed these errors:
> 
> ```
> mailserver-1  | 2024-07-29T15:28:40.633135+02:00 mail dovecot: auth: Error: passwd-file /etc/dovecot/userdb:User postmaster@mydomain.com has invalid UID '0'
> mailserver-1  | 2024-07-29T15:28:40.633154+02:00 mail dovecot: auth: Error: passwd-file /etc/dovecot/userdb:User test@mydomain.com has invalid UID '0'
> mailserver-1  | 2024-07-29T15:28:40.633164+02:00 mail dovecot: auth: Error: passwd-file /etc/dovecot/userdb:User test2@mydomain.com has invalid UID '0'
> ```
> 
> But the files are there and they have the expected data:
> 
> ```
> root@mail:/# cat /etc/dovecot/userdb | wc -l
> 3
> ```

These appear to be due to configuration decisions on your end.

---

> > Could you please confirm that your issue still occurs with [`compose.yaml` like with our example](https://github.com/docker-mailserver/docker-mailserver/blob/34423c2f665b8c7839c40fbc7d1fe8cda6342456/compose.yaml), no Docker Swarm involved.
> 
> Yes, it does.

I would suggest without any `volumes`, but first without any separate ENV file. I tend to discourage the ENV file as users report with that with a copy/paste including all defaults. It's better when only explicit changes are provided, and for that you might as well define the ENV in the `compose.yaml`?

The `compose.yaml` linked example should work fine if you don't use your modified ENV file.

---

> > Since this is just about `doveadm auth test` as your reproduction verifier, you really shouldn't need much for a minimal reproduction:
> > ```
> > $ docker run --rm -it --hostname mail.example.test --name dms mailserver/docker-mailserver:edge
> > 
> > # In another terminal, shell into the DMS container to attempt reproduction:
> > $ docker exec -it dms bash
> > $ setup email add hello@example.test secret
> > $ doveadm auth test hello@example.test secret
> > passdb: hello@example.test auth succeeded
> > extra fields:
> >   user=hello@example.test
> > ```
> 
> This actually succeded

Good, that is expected. The equivalent `compose.yaml` example shared in an earlier comment above should also work. As hinted already, problem seems to be your custom ENV choices.

---

> And now I'm out of options, besides my original one: there's something wrong in my env config ìbut I don't have the expertise on DMS to figure it out by reading it, like I would with something I'm familiar with.

Well... that's why we have our docs. There's a dedicated ENV page there that should provide more context on what is relevant and you should only be changing the stuff you understand is actually required.

For example, you shared your ENV (not in this issue but the original discussion), and we can see that `DMS_VMAIL_UID` (and the GID equivalent) have been set to `0` from the default `5000`. It's likely that's causing the `passwd-file` errors about `invalid UID '0'`.

That may be a bug, the [feature was introduced because a user needed to change it from our default](https://github.com/docker-mailserver/docker-mailserver/pull/3550) (_due to NFS_), but I don't know if it's compatible with root, do you know why you need it? (_the feature was for changing UID/GID to above 10000_)

`SSL_DOMAIN=mydomain.com` - don't do this, remove it. That is not even in the ENV file. It is still in our TLS docs for Traefik, but AFAIK is no longer relevant.

**Comment by polarathene** [comment]
**TL;DR:** Doesn't seem like Dovecot supports UserDB with root UID assignment. Why do you need that?

---

> So the most likely culprit here is my env configuration, isn't it? (reason why I posted it all at the beginning)
> This project has a ton of options (not a bad thing, just pretty incovienient to test the all when one is new to a project), what would be a minum viable env I can use to see if at least with that works.

Avoid Docker Swarm and k8s/k3s for now. You are troubleshooting, you need to identify what specific change you've done that introduced the problems.

I've provided you with the minimal example that you've verified actually works properly. So your task now is to add to that until you find it fails. Use the equivalent `compose.yaml` shared in the earlier comment. You can probably just add the ENV file and reproduce your problem.

As for advice on how to make this troubleshooting step easier. You either start from the minimal config, or your existing one, and you progress towards the other. You don't have to do this one setting at a time, you could add/remove 50% of it, then check if it broke/unbroke and if there was no change, adjust until there is (75% this time).

If the min config broke, or the real config unbroke (_whichever way you choose to go_), then you undo that last change you made and repeat the same steps but going the opposite way (_if the change was at 50%, then reduce it to 25%, if it was at 75% then you know it's something in that 50-75% change set_).

The biggest change of course is going to be from `volumes` and `env_file`. Just commenting out `env_file` will probably avoid the issue, then you can just focus on removing half of the ENV there, and then adding back or removing another half of the ENV file content.

That shouldn't take too long to pin it down. It's probably the VMAIL ENV?

---

**EDIT:** Yeah here you go simple:

```console
$ docker run --rm -itd --hostname mail.example.test --name dms --env DMS_VMAIL_UID=0 mailserver/docker-mailserver:14.0
$ docker exec -it dms setup email add hello@example.test secret
$ docker exec -it dms doveadm auth test hello@example.test secret
passdb: hello@example.test auth failed
extra fields:
  user=hello@example.test

# Last two lines:
$ docker logs dms
2024-07-29T23:07:13.228967+00:00 mail dovecot: auth: Error: passwd-file /etc/dovecot/userdb:User hello@example.test has invalid UID '0'
2024-07-29T23:07:13.229532+00:00 mail dovecot: auth: passwd-file(hello@example.test): unknown user (SHA1 of given password: e5e9fa)
```

And that would be due to the [caveat mentioned here](https://doc.dovecot.org/admin_manual/system_users_used_by_dovecot/#uids) in Dovecot docs for the UID, you would need to lower [`first_valid_uid`](https://doc.dovecot.org/settings/core/#core_setting-first_valid_uid) which is `500` by default for security reasons.

```console
$ docker exec -it dms doveconf first_valid_uid
first_valid_uid = 500
```

You could use our [`user-patches.sh` feature](https://docker-mailserver.github.io/docker-mailserver/v14.0/config/advanced/override-defaults/user-patches/) to modify that, there is also [Dovecot override support](https://docker-mailserver.github.io/docker-mailserver/v14.0/config/advanced/override-defaults/dovecot/#override-configuration) which works by providing a `dovecot.cf` file in the config volume like:

```
first_valid uid = 123
```

While you will get an updated `doveconf` response, this won't help as it turns out the `DMS_VMAIL_UID` feature actually failed, and this was present in the log output (_your shared log in the discussion however was missing this for some reason_):

```
usermod: UID '0' already exists
```

That's from when we try to change the UID for the `docker` user. So the `docker` UID remains at `5000` but the Dovecot UserDB gets updated with the new UID and there's the problem.

If the UID exists already, you can't assign it to more than one user. That said when using another UID that is taken like `8` (the `mail` system user), while the `docker` user is not updated due to this UID conflict, the `doveadm` auth does not fail (_I have set `first_valid_uid = 0`, but it seems to be valid even without that for some reason_). So it's probably that `root` / `0` is just not supported by Dovecot.

**Comment by polarathene** [comment]
Another scenario I've run into recently was if the `/etc/dovecot/userdb` config has usernames (email address) configured with any uppercase values. Dovecot is configured by default to normalize the username provided at login to lower case, so technically the entry in UserDB should also be lowercase otherwise it'll fail to match.

Same concern can be run into via LDAP/OAuth2 if those sources provide a login username that is a mismatch in casing to the account lookup.

---

@unlucio let us know when you can if the problem has been correctly identified and resolved, thanks :)
